lib/matplotlib/colorbar.py

```diff
@@ -621,37 +617,27 @@ def _add_solids_patches(self, X, Y, C, mappable):
 
     def _do_extends(self, extendlen):
         """
-        Make adjustments of the inner axes for the extend triangles (or
-        rectanges) and add them as patches.
+        Add the extend tri/rectangles on the outside of the axes.
         """
         # extend lengths are fraction of the *inner* part of colorbar,
         # not the total colorbar:
         elower = extendlen[0] if self._extend_lower() else 0
         eupper = extendlen[1] if self._extend_upper() else 0
-        total_len = eupper + elower + 1
-        elower = elower / total_len
-        eupper = eupper / total_len
-        inner_length = 1 / total_len
-
-        # make the inner axes smaller to make room for the extend rectangle
-        top = elower + inner_length
+        top = eupper + 1
```

Comment:
For consistency I'd suggest either defining `bot = -elower` here and using that below or using `1 + eupper` in the calculations.